src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/resources/viewmodel/ResourcesViewModel.kt

```diff
@@ -0,0 +1,46 @@
+package org.wycliffeassociates.otter.jvm.app.ui.resources.viewmodel
+
+import io.reactivex.Observable
+import javafx.application.Platform
+import javafx.beans.property.SimpleObjectProperty
+import javafx.beans.property.SimpleStringProperty
+import org.wycliffeassociates.otter.common.data.workbook.*
+import org.wycliffeassociates.otter.jvm.app.widgets.resourcecard.model.ResourceGroupCardItemList
+import org.wycliffeassociates.otter.jvm.app.widgets.resourcecard.model.resourceGroupCardItem
+import tornadofx.*
+
+class ResourcesViewModel : ViewModel() {
+    val activeWorkbookProperty = SimpleObjectProperty<Workbook>()
+    val workbook: Workbook
+        get() = activeWorkbookProperty.value
+
+    val activeChapterProperty = SimpleObjectProperty<Chapter>()
+    val chapter: Chapter
+        get() = activeChapterProperty.value
+
+    val activeResourceSlugProperty = SimpleStringProperty()
+    val resourceSlug: String
+        get() = activeResourceSlugProperty.value
+
+    val resourceGroups: ResourceGroupCardItemList = ResourceGroupCardItemList(mutableListOf())
+
+    fun <T, R:Any> Observable<T>.mapNotNull(f: (T) -> R?): Observable<R> =
+        concatMapIterable { listOfNotNull(f(it)) }
+
+    fun loadResourceGroups() {
+        chapter
+            .children
+            .startWith(chapter)
+            .mapNotNull { resourceGroupCardItem(it, resourceSlug, onSelect = this::navigateToTakesPage) }
+            .buffer(2) // Buffering by 2 prevents the list UI from jumping while groups are loading
```

Comment:
what happens if there's an odd number of resource groups?

Reply:
When the source observable completes, the observable will emit whatever is in the current buffer, so it won't matter if there's an odd number of resource groups.